GCXMulticastDNSKit/Discovery.swift

```diff
@@ -285,6 +286,28 @@ extension Discovery: NetServiceBrowserDelegate {
     public func netServiceBrowser(_ browser: NetServiceBrowser, didNotSearch errorDict: [String: NSNumber]) {
         guard let item = item(serviceBrowser: browser) else { return }
 
+        if let errorCode = errorDict[NetService.errorCode] as? Int {
+            if let netServiceError = NetService.ErrorCode(rawValue: errorCode) {
+                switch netServiceError {
+                // when unknown error is received, browsing has stopped and must be restarted.
+                // this happens when the app suspends/resumes
+                case .unknownError:
+                    // notify the caller that all existing discovered services have dissapeared
+                    for service in item.netServices {
+                        notifyDiscoveryServiceDidDisappear(service: DiscoveryService(configuration: item.configuration, netService: service))
+                    }
+                    // immediately restart discovery
+                    self.startDiscovery()
+
+                    // don't send .browsingFailure
+                    return
```

Comment:
Seems the right thing to do in our own projects context but not sure if we should do it in this opaque way in this public  framework.
@PaulEhrhardt @cnetthoefel Any opinions on this?
Is it okay to just restart scanning internally if we add some public documentation for this behavior?

@james-hutchby-gcx 
Maybe we should add a test-case to the unit-tests for this?

Reply:
@janrose-gcx 
It would be possible to handle this in the apps instead (catch the error, clear discoveries and restart scanning)..
But I can't see why any user of the library wouldn't want this change.

If we don't merge this PR, we should document it as a KNOWN ISSUE and include the workaround in the example because it's unexpected behaviour and difficult to understand and debug.

Reply:
I agree, for me it looks like you found a solution for a bug that would also affect users of this lib so i would be in favour of merging this PR and creating a new release including this fix.
If problems arise users still can fall back to the current version and submit an [issue](https://github.com/grandcentrix/GCXMulticastDNSKit/issues).